Isolate Sinopia server operations to a single module

mjgiarlo opened the pull request:
This is a refactoring commit with the following pieces included:
* Remove knowledge of Sinopia server internals, including authentication, from `ImportResourceTemplate` component
* Extract `sinopiaServerSpoof` module from `sinopiaServer` to disentangle them
* Drop unnecessary `.js[x]` extension from `import` statements for consistency

jermnelson commented on __tests__/sinopiaServerSpoof.test.js:
```diff
@@ -0,0 +1,94 @@
+import Config from '../src/Config'
```
Needs copyright statement.

mjgiarlo replied:
Added.

jermnelson commented on src/sinopiaServerSpoof.js:
```diff
@@ -0,0 +1,101 @@
+const barcodeRt = require('../static/spoofedFilesFromServer/fromSinopiaServer/resourceTemplates/Barcode.json')
```
Needs copyright statement.

mjgiarlo replied:
Added.

mjgiarlo commented:
@jermnelson I've added the missing copyright statement to the files touched in my branch earlier *and* all the other files that were missing it. I also corrected the package.json so that it asserts the right license.
